str/associatr/meta_analysis/run_gene_level_pvals_meta.py

```diff
@@ -0,0 +1,276 @@
+
+    Returns:
+        pvals, gene_name, row_dict
+    """
+    import pandas as pd
+
+    # read the raw results
+    gene_results = pd.read_csv(gene_file, sep='\t')
+    pvals = gene_results.iloc[:, 7]  # stored in the 8th column
+    # Find and store the attributes of the locus with lowest raw pval
+    # Find the minimum value in column 8
+    min_value = gene_results.iloc[:, 7].min()
+    # Find the rows with the minimum value in column 8
+    min_rows = gene_results[gene_results.iloc[:, 7] == min_value]
+
+    # create a dictionary of {key: list}
+    row_dict: dict[str, list] = {key: [] for key, value in VALUES_TO_INDEXES}
+
+    # populate the dict
+    for _index, row in min_rows.iterrows():
+        for key, value in VALUES_TO_INDEXES:
+            row_dict[key].append(row.iloc[value])
+
+    pvals = np.array(pvals)
+    gene_name = gene_file.split('/')[-1].split('_')[0]
+    return pvals, gene_name, row_dict
+
+
+def cct(gene_files: list[str], cell_type: str, chromosome: str, og_weights=None):
+    """
+    takes a list of gene files to process in this job
+    moves processing out of the driver job
+
+    weights were breaking - this script never supplies a value to weights, but
+    the looping caused weights to be taken from the previous round's value
+
+    Args:
+        gene_files (list): the list of files to process
+        cell_type (str):
+        chromosome (str):
+        og_weights (): None
+
+    Returns:
+
+    """
+    """
+    Code adapted from the STAR package https://github.com/xihaoli/STAAR/blob/dc4f7e509f4fa2fb8594de48662bbd06a163108c/R/CCT.R wtih a modifitcaiton: when indiviudal p-value = 1, use minimum p-value
+    #' An analytical p-value combination method using the Cauchy distribution
+    #'
+    #' The code{CCT} function takes in a numeric vector of p-values, a numeric
+    #' vector of non-negative weights, and return the aggregated p-value using Cauchy method.
+    #' @param pvals a numeric vector of p-values, where each of the element is
+    #' between 0 to 1, to be combined.
+    #' @param weights a numeric vector of non-negative weights. If code{NULL}, the
+    #' equal weights are assumed.
+    #' @return the aggregated p-value combining p-values from the vector code{pvals}.
+    #' @examples pvalues <- c(2e-02,4e-04,0.2,0.1,0.8)
+    #' @examples CCT(pvals=pvalues)
+    #' @references Liu, Y., & Xie, J. (2020). Cauchy combination test: a powerful test
+    #' with analytic p-value calculation under arbitrary dependency structures.
+    #' emph{Journal of the American Statistical Association 115}(529), 393-402.
+    #' (href{https://www.tandfonline.com/doi/full/10.1080/01621459.2018.1554485}{pub})
+    #' @export
+    R code is implemented in python
+    """
+    # Import here as a PythonJob's function must stand alone
+    from scipy.stats import cauchy  # noqa: PLC0415
+
+    from cpg_utils import to_path
+    from cpg_utils.config import output_path
+
+    for gene_file in gene_files:
```

Comment:
possible check here - if output file already exists, skip onto the next gene

This has happened before when the jobs are preempted/restarted, and it starts from the beginning of the gene list again. Running with full/test permissions will mean that it doesn't fail, but avoiding the repeated loops should mean you can run using standard permissions